src/features/import/hooks/useDateConfig.ts

```diff
@@ -33,7 +31,7 @@ export default function useDateConfig(column: DateColumn, columnIndex: number) {
     }
 
     if (column.dateFormat) {
-      return !parseDate(value, column.dateFormat);
+      return !dateParsingIsValid(value, column.dateFormat);
```

Comment:
I started walking along this road now, but realised that it's difficult because of the `SwedishPnrParser`: creating the `AnyFormatDateParser` in the constructor works for danish and norwegian person numbers, because they always have the same format. But swedish ones can have either 2 or 4 digit years, which and we can't know which unless we take in the value. 
So to do this refactor (that i think makes a lot of sense to do) i think the `SwedishPnrParser` needs to somehow work differently...?

Reply:
Good catch! Well, you could still do this refactor (as suggested specifically in this comment) because `SwedishPnrParser` could still create the "child parser" in `parse()` instead of doing it in the constructor.

But the other option would be for `SwedishPnrParser` to create two "child parsers", one called `eightDigitParser` and one called `sixDigitParser`, and then conditionally use one or the other in the methods.